tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/OverlayAccess.java

```diff
@@ -0,0 +1,327 @@
+		unselect(SearchOptions.CASE_SENSITIVE);
+		unselect(SearchOptions.WHOLE_WORD);
+	}
+
+	@Override
+	public void closeAndRestore() {
+		restoreInitialConfiguration();
+		assertInitialConfiguration();
+		closeOperation.run();
+	}
+
+	@Override
+	public void close() {
+		closeOperation.run();
+	}
+
+	@Override
+	public void select(SearchOptions option) {
+		ToolItem button= getButtonForSearchOption(option);
+		if (button == null) {
+			return;
+		}
+		button.setSelection(true);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(false);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void unselect(SearchOptions option) {
+		ToolItem button= getButtonForSearchOption(option);
+		if (button == null) {
+			return;
+		}
+		button.setSelection(false);
+		if (option == SearchOptions.GLOBAL) {
+			button.setSelection(true);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void simulateEnterInFindInputField(boolean shiftPressed) {
+		simulateKeyPressInFindInputField(SWT.CR, shiftPressed);
+	}
+
+	@Override
+	public void simulateKeyPressInFindInputField(int keyCode, boolean shiftPressed) {
+		final Event event= new Event();
+		event.type= SWT.KeyDown;
+		event.keyCode= keyCode;
+		if (shiftPressed) {
+			event.stateMask= SWT.SHIFT;
+		}
+		find.notifyListeners(SWT.KeyDown, event);
+		find.traverse(SWT.TRAVERSE_RETURN, event);
+		FindReplaceTestUtil.runEventQueue();
+	}
+
+	@Override
+	public String getFindText() {
+		return find.getText();
+	}
+
+	@Override
+	public String getReplaceText() {
+		return replace.getText();
+	}
+
+	@Override
+	public void setFindText(String text) {
+		find.setText(text);
+		find.notifyListeners(SWT.Modify, null);
+	}
+
+	@Override
+	public void setReplaceText(String text) {
+		openReplaceDialog();
+		replace.setText(text);
+	}
+
+	@Override
+	public ToolItem getButtonForSearchOption(SearchOptions option) {
+		switch (option) {
+			case CASE_SENSITIVE:
+				return caseSensitive;
+			case REGEX:
+				return regEx;
+			case WHOLE_WORD:
+				return wholeWord;
+			case GLOBAL:
+				return inSelection;
+			//$CASES-OMITTED$
+			default:
+				return null;
+		}
+	}
+
+	private Set<SearchOptions> getEnabledOptions() {
+		return Arrays.stream(SearchOptions.values())
+				.filter(option -> {
+					ToolItem buttonForSearchOption= getButtonForSearchOption(option);
+					if (option == SearchOptions.GLOBAL) {
+						return buttonForSearchOption.getEnabled();// The "Global" option is mapped to a button that
+						// selects whether to search in the selection, thus inverting the semantic
+					}
+					return buttonForSearchOption != null && buttonForSearchOption.getEnabled();
+				})
+				.collect(Collectors.toSet());
+	}
+
+	private Set<SearchOptions> getSelectedOptions() {
+		return Arrays.stream(SearchOptions.values())
+				.filter(isOptionButtonSelected())
+				.collect(Collectors.toSet());
+	}
+
+	private Predicate<? super SearchOptions> isOptionButtonSelected() {
```

Comment:
Nitpicky: the method name does not reflect what the method does. It is not about whether a button is selected, but whether the option that is toggled by the button is currently activated, For example, if the button for the global option is selected, this method returns `false`.

Reply:
I agree, this method models the enablement state of the option, not of the button (as it should!). I have renamed it to `isOptionSelected()`

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/2f9136a25069244769073ee77ad5be15b5dadeee